vignettes/communicate.Rmd

```diff
@@ -0,0 +1,349 @@
+---
+title: "Communicate lifecycle changes in your functions"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Communicate lifecycle changes in your functions}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+
+options(
+  # Pretend we're in the lifecycle package
+  "lifecycle:::calling_package" = "lifecycle",
+  # suppress last_warnings() message by default
+  "lifecycle_verbosity" = "warning"
+)
+```
+
+lifecycle provides a standard way to document the lifecycle stages of functions and arguments, paired with tools to steer users away from deprecated functions.
+Before we go in to the details, make sure that you're familiar with the lifecycle stages as described in `vignette("stages")`.
+
+## Basics
+
+lifecycle badges make it easy for users to see the lifecycle stage when reading the documentation.
+To use the badges, first call `usethis::use_lifecycle()` to embed the badge images in your package (you only need to do this once), then use `lifecycle::badge()` to insert a badge:
+
+```{r, eval = FALSE}
+#' `r lifecycle::badge("experimental")
+#' `r lifecycle::badge("deprecated")
+#' `r lifecycle::badge("superseded")
+```
+
+Deprecated functions also need to advertise their status when run.
+lifecycle provides `deprecate_warn()` which takes three main arguments:
+
+-   The first argument, `when`, gives the version number when the deprecation occurred.
+
+-   The second argument, `what`, describes exactly what was deprecated.
+
+-   The third argument, `with`, provides the recommended alternative.
+
+We'll cover the details shortly, but here are a few sample uses:
+
+```{r}
+lifecycle::deprecate_warn("1.0.0", old_fun(), new_fun())
+lifecycle::deprecate_warn("1.0.0", fun(), testthat::fun())
+lifecycle::deprecate_warn("1.0.0", fun(old_arg), fun(new_arg))
+```
```

Comment:
The `lifecycle:::calling_package` option doesn't seem to be working exactly. The intent is, I think, to see "lifecycle" there but instead we're seeing "base".

```
lifecycle::deprecate_warn("1.0.0", old_fun(), new_fun())
#> Warning: `old_fun()` was deprecated in base 1.0.0.
#> Please use `new_fun()` instead.
```

Reply:
Yeah, just ignore that for now.